File: app_manager/appmanager/src/main/java/app/editors/manager/ui/views/edits/BaseInputFilter.java
package app.editors.manager.ui.views.edits;

import android.text.InputFilter;
import android.text.Spanned;

public class BaseInputFilter implements InputFilter {

    protected String mResultString;

    @Override
    public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
        final StringBuilder resultStr = new StringBuilder(dest);
        resultStr.replace(dstart, dend, source.toString());
        mResultString = resultStr.toString();
        return null;
    }

}

File: toolkit/libtoolkit/src/main/java/lib/toolkit/base/ui/views/edits/BaseInputFilter.kt
package lib.toolkit.base.ui.views.edits

import android.text.InputFilter
import android.text.Spanned

open class BaseInputFilter : InputFilter {

    protected var mResultString: String = ""

    override fun filter(source: CharSequence, start: Int, end: Int, dest: Spanned, dstart: Int, dend: Int): CharSequence? {
        val resultStr = StringBuilder(dest)
        resultStr.replace(dstart, dend, source.toString())
        mResultString = resultStr.toString()
        return null
    }

}
